src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -2411,6 +2413,27 @@ def update_ddos_plan(cmd, resource_group_name, ddos_plan_name, tags=None, vnets=
 
 # region DNS Commands
 # add delegation name server record for the created child zone in it's parent zone.
+def _to_snake(s):
+    import re
+    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', s)
+
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+
+def _convert_to_snake_case(element):
+    if isinstance(element, dict):
+        ret = dict()
+        for k, v in element.items():
+            ret[_to_snake(k)] = _convert_to_snake_case(v)
+
+        return ret
+
+    if isinstance(element, list):
+        return [_convert_to_snake_case(i) for i in element]
+
+    return element
```

Comment:
What if the element contains property like `tags`? This will change the key from camel case to snack case.

Reply:
There is no properties like that, all properties are used for checking: https://github.com/Azure/azure-cli/blob/478f69ce135478ad138bb503f6f645df2df2eea9/src/azure-cli/azure/cli/command_modules/network/custom.py#L2985-L3045